### ISSUE TITLE
RT estimation for multislope decay

### ISSUE THREAD

**Issue by orchidas** [artificial-audio/pyFDN#203]
Currently, estmate_rt_bands in auxilliary.acoustics assumes single slope decay. It would be useful to use DecayFitNet as a dependency to estimate multiple slopes from a multislope EDC.

**Comment by SebastianJiroSchlecht** [comment]
I agree that it would be useful to have this. Would you like to implement it?

I'm researching at the moment what the best way is to integrate dependencies that are only used occasionally.

**Comment by orchidas** [comment]
The dependency is the issue - do we add DecayFitNet as a submodule? Once we decide that, I can implement it. We also have the whole common slopes pipeline implemented in [slope2noise](https://github.com/gdalsanto/slope2noise). 

For my personal work, I created a [fork of DecayFitNet](https://github.com/orchidas/DecayFitNet) with a pyproject.toml which can be installed using `poetry`. In that case, you simply add the dependency to pyFDN's pyproject.toml, and `poetry install` takes care of the rest. I would also suggest using `poetry` for installing pyFDN. The `pyproject.toml` will need to be modified.

**Comment by SebastianJiroSchlecht** [comment]
## Proposed dependency boundary

Keep decay estimation providers out of the pyFDN runtime unless they are genuinely core functionality.

For this issue, use the following design:

1. **Keep a lightweight single-slope estimator in pyFDN**, but implement it using the existing NumPy/SciPy stack instead of `pyroomacoustics`.
   - Generate octave-band edges/centre frequencies locally.
   - Build the Butterworth SOS filter bank with `scipy.signal.butter`.
   - Filter with `scipy.signal.sosfilt`.
   - Estimate RT from the Schroeder energy decay curve and a documented linear-fit range.
   - Keep `estimate_initial_level_bands` on the same SciPy filter bank.
2. **Remove `pyroomacoustics` from pyFDN's mandatory dependencies.** Its remaining use is room simulation in an example, so declare it only in an example-specific repository dependency group.
3. **Use DecayFitNet directly from the multi-slope example.** Do not add a DecayFitNet adapter or import to `src/pyFDN`, and do not expose DecayFitNet-specific result objects through the pyFDN API. Convert its results to ordinary NumPy arrays before passing them to pyFDN synthesis/design functions.
4. Use repository-only dependency groups for heavy example dependencies, for example:

```toml
[dependency-groups]
examples = [
    "marimo",
    "matplotlib>=3.5",
    "plotly>=5",
]
example-room-simulation = [
    { include-group = "examples" },
    "pyroomacoustics",
]
example-multislope = [
    { include-group = "examples" },
    "decayfitnet",
]
```

A package extra can be added later if running these examples from a published pyFDN installation becomes a supported user-facing workflow.

## Making DecayFitNet installable from PyPI

Because we maintain DecayFitNet, the preferred solution is to package and release it independently rather than use a git submodule or vendor it into pyFDN.

### 1. Give it a conventional package layout

The current import name `toolbox` is too generic. Move the reusable inference code under a project-specific namespace and keep demos/training code outside the runtime package:

```text
DecayFitNet/
├── pyproject.toml
├── README.md
├── LICENSE
├── src/
│   └── decayfitnet/
│       ├── __init__.py
│       ├── inference.py
│       ├── preprocessing.py
│       ├── model.py
│       └── models/
│           ├── __init__.py
│           ├── DecayFitNet_v10.onnx
│           ├── DecayFitNet_1slopes_v10.onnx
│           ├── DecayFitNet_2slopes_v10.onnx
│           ├── DecayFitNet_3slopes_v10.onnx
│           └── input transforms
├── tests/
└── examples/
```

Expose a small stable API from `decayfitnet/__init__.py`, for example `DecayFitNet` and/or `estimate_decay`. Avoid requiring users to import internal modules.

### 2. Add complete PEP 621 metadata

A minimal setuptools-based `pyproject.toml` could be:

```toml
[build-system]
requires = ["setuptools>=77", "wheel"]
build-backend = "setuptools.build_meta"

[project]
name = "decayfitnet"
version = "0.1.0"
description = "Neural-network estimation of multi-exponential sound energy decay"
readme = "README.md"
requires-python = ">=3.10"
license = "MIT"
authors = [
    { name = "Georg Götz" },
    { name = "Ricardo Falcón Pérez" },
    { name = "Sebastian J. Schlecht" },
    { name = "Ville Pulkki" },
]
classifiers = [
    "Development Status :: 3 - Alpha",
    "Intended Audience :: Science/Research",
    "Programming Language :: Python :: 3",
]
dependencies = [
    "numpy",
    "scipy",
    "torch",
    "torchaudio",
    "onnxruntime",
]

[project.optional-dependencies]
plot = ["matplotlib"]
train = ["h5py", "onnx", "tensorboard"]
test = ["pytest", "build", "twine"]

[project.urls]
Repository = "https://github.com/georg-goetz/DecayFitNet"
Issues = "https://github.com/georg-goetz/DecayFitNet/issues"

[tool.setuptools]
package-dir = { "" = "src" }

[tool.setuptools.packages.find]
where = ["src"]

[tool.setuptools.package-data]
decayfitnet = ["models/*.onnx", "models/*.pkl"]
```

The exact runtime list must be verified from imports. In particular, plotting, dataset generation, evaluation, and training modules should not be imported by the inference package, otherwise `matplotlib` and `h5py` remain mandatory. The current ONNX inference path can likely use `onnxruntime.InferenceSession` directly; if model validation is not required at runtime, `onnx` itself can remain a training/export extra.

A later preprocessing refactor to NumPy/SciPy could potentially remove PyTorch and Torchaudio from inference, but that is not required for the first correct release.

### 3. Treat model files as package resources

Do not locate models by walking parent directories from `__file__`. Load them with `importlib.resources`, which works with an installed wheel:

```python
from importlib.resources import as_file, files
import onnxruntime as ort

resource = files("decayfitnet.models").joinpath("DecayFitNet_v10.onnx")
with as_file(resource) as model_path:
    session = ort.InferenceSession(str(model_path))
```

For the initial wheel, include only the ONNX models actually used for inference (for example, the four v10 variants), not duplicate v9 and `.pth` training checkpoints. The existing ONNX files are only about 2.6–2.7 MB each, so bundling the selected models in the wheel is reasonable. Keep model provenance/version information in the package and confirm that the repository's MIT license covers distribution of the weights.

The pickle transform files can be included initially, but a non-executable format such as JSON or NPZ would be safer and easier to inspect in a future release.

### 4. Make the installed API independent of the repository layout

Add an inference smoke test that:

- installs the built wheel into a clean environment;
- imports `decayfitnet` from outside the repository;
- constructs the estimator;
- locates every packaged model/resource;
- runs a small CPU-only inference;
- verifies output shapes for one-, two-, three-, and automatically selected slopes.

This catches the common case where editable installs work but wheels fail because model paths or package data are wrong.

### 5. Build and validate locally

```bash
python -m pip install --upgrade build twine
python -m build
python -m twine check dist/*
python -m pip install --force-reinstall dist/*.whl
```

Also inspect the wheel contents (`unzip -l dist/*.whl`) and run the test suite in the clean environment. Test the supported Python versions and at least Linux/macOS CPU inference before publishing.

### 6. Publish safely

- Check that the normalized PyPI project name `decayfitnet` is available.
- Publish the first candidate to TestPyPI and install it from there.
- Prefer PyPI Trusted Publishing from a GitHub Actions release workflow instead of storing a long-lived API token.
- Tag the exact tested commit (for example `v0.1.0`) and publish only from that tag.
- Once released, pin a compatible lower bound in the example dependency group, e.g. `decayfitnet>=0.1.0`.

## Acceptance criteria

- [ ] `pip install pyFDN` no longer installs `pyroomacoustics`.
- [ ] `estimate_rt_bands` and `estimate_initial_level_bands` work using NumPy/SciPy only.
- [ ] Existing synthetic-decay tests pass without `pyroomacoustics`.
- [ ] The room-simulation example declares `pyroomacoustics` only in its example environment.
- [ ] DecayFitNet has a versioned, clean-installable PyPI wheel containing the required inference models.
- [ ] A multi-position/multi-slope example imports DecayFitNet directly and passes estimator-neutral arrays into pyFDN.
- [ ] Core CI runs without optional example dependencies; a separate example/integration job exercises DecayFitNet.

**Comment by SebastianJiroSchlecht** [comment]
@orchidas 
We could give the "Making DecayFitNet installable from PyPI" issue to one of our assistants (or would you like to do it)?
@jeremybbq : maybe this is interesting for you as well?
